src/ragas/prompts/base.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import annotations
+
+import json
+import typing as t
+
+from langchain.prompts import ChatPromptTemplate, HumanMessagePromptTemplate
+from langchain_core.prompt_values import PromptValue
+from langchain_core.messages import BaseMessage
+
+try:
+    from pydantic.v1 import root_validator
+except ImportError:
+    from pydantic import root_validator
+
+class RagasPrompt(PromptValue):
+    """
+    RagasPrompt is a class that represents a prompt for the ragas metrics.
+    """
+    instruction: str
+    examples: t.List[t.Dict[str, t.Any]] = []
+    input_keys: t.List[str]
+    output_key: str
+    output_type: str = 'JSON'
+    
+    def to_string(self) -> str:
+        """Return prompt value as string."""
+        ...
+    
+    def to_messages(self) -> t.List[BaseMessage]:
+        """Return prompt as a list of Messages."""
+        ...
```

Comment:
do implement these as well. I'm changing the RagasLLM to use this extensively so this would be the primary way of using `Prompt` internally

Reply:
generate_prompt_string function converts to string. I will change that to to_string and implement to_messages